server/api/server.py

```
@@ -19,12 +30,25 @@
+def handle_incorrect_oauth_callback():
+    from flask import redirect, request
+    # Redirect to Next.js frontend route handler with all query params
+    frontend_url = f"http://localhost:3000/api/auth/google/callback?{request.query_string.decode()}"
+    return redirect(frontend_url, code=302)
+
+
 if __name__ == "__main__":
-    app.run(debug=True)
+    # Initialize database tables when running directly
+    init_db()
+    app.run(debug=True)
```

Ruff (0.14.7)

Use of `debug=True` in Flask app detected

(S201)